### PR TITLE
Scrutinizer fixes

### PR DESCRIPTION
**Pull request by willmot**
Fix a bunch of minor issues raised by Scrutinizer.

See https://github.com/humanmade/backupwordpress/issues/1031

### CODE REVIEW

**Comment by willmot** [comment]
- [ ] Need to fix the new issues added by this PR before merging: https://scrutinizer-ci.com/g/humanmade/backupwordpress/inspections/43b9aa88-fbb4-4a7d-8f67-c92b520c4098/issues/

**Comment by dashaluna** [comment]
@willmot is comment https://github.com/humanmade/backupwordpress/pull/1104#issuecomment-243006112 still stands?

**Comment by willmot** [comment]
Yep

**Comment by dashaluna** [comment]
Cool, I'll wait for more commits to PR :)

**Comment by willmot** [comment]
@sambulance / @shadyvb let's merge this one, it reduces the net number of Scrutinizer issues which is a good thing.

File: admin/actions.php [hunk 1]
```diff
@@ -250,9 +248,9 @@ function edit_schedule_submit() {
 		}
 	}
 
-	if ( isset( $_POST['hmbkp_schedule_recurrence']['hmbkp_type'] ) ) {
+	$schedule_recurrence_type = sanitize_text_field( $_POST['hmbkp_schedule_recurrence']['hmbkp_type'] );
```
Comment (shadyvb) [inline admin/actions.php]: I'm not sure why we swapped the lines here, the `if` check should happen before sanitization, so as not to produce an error if the value does not exist.

File: assets/hmbkp.js [hunk 2]
```diff
@@ -139,21 +139,6 @@ var BackUpWordPressAdmin = (function($){
 			);
 		} );
 
-		jQuery( document ).one( 'click', '.hmbkp_send_error_via_email', function ( e ) {
```
Comment (shadyvb) [inline assets/hmbkp.js]: This doesn't seem like a CS/CI fix, should probably be double-checked to see if the removed code is not needed / does not need a replacement.

File: classes/backup/class-backup-status.php [hunk 3]
```diff
@@ -18,6 +19,11 @@ public function start( $backup_filename, $status_message ) {
 		$this->set_status( $status_message );
 	}
 
+	/**
+	 * Get the filename of the backup.
+	 *
+	 * @return string The backup filename.
```
Comment (shadyvb) [inline classes/backup/class-backup-status.php]: `string|null`

File: assets/hmbkp.js (hunk 2 above)
Comment (sambulance) [inline assets/hmbkp.js]: I think this was intentionally removed as unused code, I can't find a reference to either the class `hmbkp_send_error_via_email` or the function `hmbkp_email_error` in the rest of the codebase.

**Pull request merged**